manticore/core/smtlib/visitors.py

```diff
@@ -739,6 +740,10 @@ def visit_BitVecSub(self, expression, *operands):
                         taint=subright.taint | right.taint,
                     ),
                 )
+        elif isinstance(right, Constant) and right.value == 0:
+            return left
+        elif left is right:
```

Comment:
I think you mentioned this during our sync, but this checks to see if they are the same instance in memory? Is there a reason why `==` isn't used? A `==` would be more correct. Maybe add a `TODO` or some sort of comment to keep track of this reason

Reply:
I want to check that `left` and `right`  are the same. For `BitVec` objects,  `==` is overloaded to create a new equality constraint instead of actually checking for equality, so it won't work AFAIU:   
https://github.com/trailofbits/manticore/blob/abbfa01cdc790f7ccaf3c24d6726362e53f88e79/manticore/core/smtlib/expression.py#L444-L445

Reply:
Right, but then `BoolEqual` is used in the `if`, which I believe will call `BoolEqual.__bool__`, which is here https://github.com/trailofbits/manticore/blob/52007a8fa59e2234b4e69ff9167e8c35916aff6b/manticore/core/smtlib/expression.py#L210-L217

Of course, this would involve a potentially large runtime cost for the simplification.

The `is` check is better than nothing, so I'm fine leaving it like this 👍

Reply:
Oh you're right, I did miss that. `BoolEqual.__bool__` will ultimately call [visit_BoolEqual](https://github.com/trailofbits/manticore/blob/52007a8fa59e2234b4e69ff9167e8c35916aff6b/manticore/core/smtlib/visitors.py#L509) which tests for equality using `is` but handles other cases as well, so I agree that using `==` is a better alternative